File: 485-max-consecutive-ones/485-max-consecutive-ones.cpp
class Solution {
public:
/*
    Keep Counting until ones until found zeros
    If found zeros and if CountZeros > maxCount => update value of maxCount
*/
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int CountOnes = 0,maxCount = 0;
        for (int i = 0;i<nums.size();i++){
            if (nums[i] == 1) CountOnes++;
            else {
                maxCount = max(CountOnes,maxCount);
                CountOnes = 0;
            }
        }
        maxCount = max(CountOnes,maxCount);
        return maxCount;
    }
};

File: 485-max-consecutive-ones/485-max-consecutive-ones.py
class Solution:
    """
    1. Keep Counting until ones until found zeros
    2. If found zeros and if CountZeros > maxCount => update value of maxCount
    """
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        count = 0;N = len(nums)
        mx = 0
        for i in range(N):
            if nums[i] == 1:
                count += 1
            elif nums[i]== 0:
                mx = max(mx,count)
                count = 0
        mx = max(mx,count)
        return mx